auth-web/src/components/auth/manage-business/EntityManagement.vue

```diff
@@ -416,6 +443,48 @@ export default class EntityManagement extends Mixins(AccountMixin, AccountChange
 
     this.setAccountChangedHandler(this.setup)
     this.setup()
+    this.parseUrlAndAddAffiliation()
+  }
+
+  // Function to parse the URL and extract the parameters, used for magic link email
+  parseUrlAndAddAffiliation = async () => {
+    const urlParams = new URLSearchParams(window.location.search)
+    const accountId = urlParams.get('accountid')
+    const linkId = urlParams.get('linkid')
+
+    if (accountId && linkId) {
+      try {
+        const invitation = await AffiliationInvitationService.getInvitations(linkId)
+        const identifier = invitation.data?.businessIdentifier
+        // 1. link expired
+        if (invitation.data.status === AffiliationInvitationStatus.Expired) {
+          this.showLinkExpiredModal(identifier)
+          return
+        }
+
+        // 2. business already added
+        const isAdded = await this.isAffiliated(identifier)
+        const business = await BusinessService.searchBusiness(identifier)
+        if (isAdded && business && business.data && business.status === 200) {
```

Comment:
I'd almost wrap these in the service itself. Similar to what we do in `business-edit-ui`

Reply:
https://github.com/bcgov/business-edit-ui/blob/ef4a03ba0fc64f7b0297fd3aa46d9aaa39eadd18/src/services/auth-services.ts#L66 

Can still use async await and not `.then`

Reply:
I removed business, the business name should in the email link